test/test_strerror.cpp

```diff
@@ -54,3 +61,38 @@ TEST(test_strerror, get_error) {
     "' did not cause the expected error message.";
 #endif
 }
+
+/*
+   Define the blueprint of a mock identified by `strerror_r_proto`
+   strerror_r signature:
+   int strerror_r(int errnum, char *buf, size_t buflen);
+*/
+mmk_mock_define(strerror_r_mock, int, char *, size_t);
+
+/* Mocking test example */
+TEST(test_strerror, test_mock) {
+  /* Mock the strerror_r function in the current module using
+     the `strerror_r_mock` blueprint. */
+  // mmk_mock("strerror_r@rcutils", strerror_r_mock);
+  // mmk_mock("__xpg_strerror_r@self", strerror_r_mock);
+  // mmk_mock("strerror_r@self", strerror_r_mock);
+  mmk_mock("__xpg_strerror_r@lib:rcutils", strerror_r_mock);
+
+  /* Tell the mock to return NULL and set errno to ENOMEM
+     whatever the given parameter is. */
+  mmk_when(
+    strerror_r(mmk_any(int), mmk_any(char *), mmk_any(size_t) ),
```

Comment:
This sets the global behavior of the mocked function? Are there reasonable scenarios where you might want the default behavior of the mocked function generally, but have the mocked version for one specific call? How might you accomplish that?

Reply:
Yeah, that's something I thought about, but unfortunately `Mimick` mocks do not explicitly support that feature. I only know of `gMock` for a framework that can do that though.

Reply:
Uh, turns out it has the feature and @Blast545 found it :tada: